Sources/MapboxMapsSwiftUI/MapboxView.swift

```diff
@@ -0,0 +1,87 @@
+@_exported import MapboxMaps
+import SwiftUI
+
+public extension CameraState {
+    /// Initializes CameraState with center and zoom.
+    init(center: CLLocationCoordinate2D, zoom: CGFloat) {
+        self.init(center: center, padding: .zero, zoom: zoom, bearing: 0, pitch: 0)
+    }
+}
+
+/// The action that is called when the map is loaded.
+public typealias MapLoadedAction = (MapboxMap) -> Void
+
+/// View displaying Mapbox Map in SwiftUI.
+@available(iOS 13.0, *)
+public struct MapboxView: UIViewRepresentable {
+    public typealias InitialOptionsProvider = () -> MapInitOptions
+
+    @Binding
+    var camera: CameraState
+    var cameraBounds: CameraBoundsOptions?
+
+    var annotations = [PointAnnotation]()
+    var _onMapLoaded: MapLoadedAction?
+    var styleURI = StyleURI.streets
+    var getstureOptions: GestureOptions = GestureOptions()
+
+    private let initialOptions: InitialOptionsProvider?
+
+    /// Creates an instance showing scpecisif region.
+    ///
+    /// - Parameters:
+    /// - camera: The camera state to display.
+    /// - initialOptions: A closure to provide initial map parameters. It gets called only once when `MapboxView` is created.
+    public init(camera: Binding<CameraState>, initialOptions: InitialOptionsProvider? = nil) {
+        self.initialOptions = initialOptions
+        _camera = camera
+    }
+
+    public func makeCoordinator() -> SwiftUIMapViewCoordinator {
+        SwiftUIMapViewCoordinator(camera: $camera)
+    }
+
+    public func makeUIView(context: UIViewRepresentableContext<MapboxView>) -> MapView {
+        MapView(frame: .zero, mapInitOptions: initialOptions?() ?? MapInitOptions())
+    }
+
+    public func updateUIView(_ mapView: MapView, context: Context) {
+        context.environment.mapboxViewProxy?.mapView = mapView
+        context.coordinator.mapView = mapView
+        context.coordinator.update(from: self)
+    }
+}
+
+@available(iOS 13.0, *)
+extension MapboxView {
+    private func set<T>(_ keyPath: WritableKeyPath<MapboxView, T>, _ value: T) -> Self {
+        var updated = self
+        updated[keyPath: keyPath] = value
+        return updated
+    }
+
+    /// Sets camera bounds.
+    public func cameraBounds(_ cameraBounds: CameraBoundsOptions) -> Self {
+        set(\.cameraBounds, cameraBounds)
```

Comment:
why is it better than `self.cameraBounds = cameraBounds`? Is there a benefit of doing this using WritableKeyPath?